From 4f62114d8f5e4c4d293537f43d6e19d2bf5bea57 Mon Sep 17 00:00:00 2001
From: Konrad Weihmann <kweihmann@outlook.com>
Date: Sun, 10 May 2020 20:00:35 +0200
Subject: [PATCH] Makefile fixes

Signed-off-by: Konrad Weihmann <kweihmann@outlook.com>
---
 Makefile | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/Makefile b/Makefile
index dfb2a99ce..132756da5 100644
--- a/Makefile
+++ b/Makefile
@@ -77,7 +77,7 @@ else # !WINNT
         endif # !CPPCHK_GLIBCXX_DEBUG
     endif # GNU/kFreeBSD

-    LDFLAGS=-pthread
+    LDFLAGS+=-pthread

 endif # WINNT

@@ -112,11 +112,11 @@ else ifeq ($(CXX), c++)
 endif

 ifeq ($(HAVE_RULES),yes)
-    override CXXFLAGS += -DHAVE_RULES -DTIXML_USE_STL $(shell pcre-config --cflags)
+    override CXXFLAGS += -DHAVE_RULES -DTIXML_USE_STL $(shell pkg-config libpcre --cflags)
     ifdef LIBS
-        LIBS += $(shell pcre-config --libs)
+        LIBS += $(shell pkg-config libpcre --libs)
     else
-        LIBS=$(shell pcre-config --libs)
+        LIBS=$(shell pkg-config libpcre --libs)
     endif
 endif

--
2.25.1

