diff --git a/bin/genhtml b/bin/genhtml
index 2352300..d430467 100755
--- a/bin/genhtml
+++ b/bin/genhtml
@@ -284,6 +284,7 @@ our $html_epilog;	# Actual HTML epilog
 our $html_ext = "html";	# Extension for generated HTML files
 our $html_gzip = 0;	# Compress with gzip
 our $demangle_cpp = 0;	# Demangle C++ function names
+our $cppfilt = "c++filt"; # Demangling executable name
 our @opt_ignore_errors;	# Ignore certain error classes during processing
 our @ignore;
 our $opt_config_file;	# User-specified configuration file location
@@ -376,6 +377,7 @@ if ($config || %opt_rc)
 		"genhtml_charset"		=> \$charset,
 		"genhtml_desc_html"		=> \$rc_desc_html,
 		"genhtml_demangle_cpp"		=> \$demangle_cpp,
+		"genhtml_demangle_cpp_tool"	=> \$cppfilt,
 		"genhtml_missed"		=> \$opt_missed,
 		"lcov_function_coverage"	=> \$lcov_func_coverage,
 		"lcov_branch_coverage"		=> \$lcov_branch_coverage,
@@ -544,7 +546,7 @@ if ($frames)
 # Ensure that the c++filt tool is available when using --demangle-cpp
 if ($demangle_cpp)
 {
-	if (system_no_output(3, "c++filt", "--version")) {
+	if (system_no_output(3, $cppfilt, "--version")) {
 		die("ERROR: could not find c++filt tool needed for ".
 		    "--demangle-cpp\n");
 	}
@@ -5258,7 +5260,7 @@ sub demangle_list($)
 	}
 
 	# Build translation hash from c++filt output
-	open($handle, "-|", "c++filt $demangle_arg < $tmpfile") or
+	open($handle, "-|", "$cppfilt $demangle_arg < $tmpfile") or
 		die("ERROR: could not run c++filt: $!\n");
 	foreach my $func (@$list) {
 		my $translated = <$handle>;
