diff --git a/lcov_cobertura/lcov_cobertura.py b/lcov_cobertura/lcov_cobertura.py
index 379e405..f0a90d2 100755
--- a/lcov_cobertura/lcov_cobertura.py
+++ b/lcov_cobertura/lcov_cobertura.py
@@ -23,10 +23,6 @@ from distutils.spawn import find_executable
 __version__ = '2.0.2'
 
 CPPFILT = "c++filt"
-HAVE_CPPFILT = False
-
-if find_executable(CPPFILT) is not None:
-    HAVE_CPPFILT = True
 
 
 class Demangler():
@@ -387,6 +383,7 @@ def main(argv=None):
     """
     if argv is None:
         argv = sys.argv
+    global CPPFILT
     parser = OptionParser()
     parser.usage = ('lcov_cobertura.py lcov-file.dat [-b source/dir] '
                     '[-e <exclude packages regex>] [-o output.xml] [-d]')
@@ -403,11 +400,20 @@ def main(argv=None):
     parser.add_option('-d', '--demangle',
                       help='Demangle C++ function names using %s' % CPPFILT,
                       action='store_true', dest='demangle', default=False)
+    parser.add_option('-t', '--demangle-tool',
+                      help='Program to use demangling c++function name',
+                      action='store', dest="cppfilt", default="c++filt")
     parser.add_option('-v', '--version',
                       help='Display version info',
                       action='store_true')
     (options, args) = parser.parse_args(args=argv)
 
+    CPPFILT = options.cppfilt
+
+    HAVE_CPPFILT = False
+    if find_executable(CPPFILT) is not None:
+        HAVE_CPPFILT = True
+
     if options.demangle and not HAVE_CPPFILT:
         raise RuntimeError("C++ filter executable (%s) not found!" % CPPFILT)
     if options.version:
