From 1c02d40ce1b2adf59365e408aa2096ec90c84d8e Mon Sep 17 00:00:00 2001
From: Sung Gon Kim <sunggon82.kim@lge.com>
Date: Sun, 21 Feb 2021 12:15:32 +0000
Subject: [PATCH] #269 Fixes an issue that clang-tidy causes compilation

---
 Source/cmcmd.cxx | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/Source/cmcmd.cxx b/Source/cmcmd.cxx
index 67394f90..11b363e7 100644
--- a/Source/cmcmd.cxx
+++ b/Source/cmcmd.cxx
@@ -381,6 +381,12 @@ int HandleTidy(const std::string& runCmd, const std::string& sourceFile,
     cm::append(tidy_cmd, orig_cmd);
   }
 
+  std::string target_sys;
+  if (cmSystemTools::GetEnv("TARGET_SYS", target_sys)) {
+    tidy_cmd.emplace_back("-target");
+    tidy_cmd.emplace_back(target_sys);
+  }
+
   // Run the tidy command line.  Capture its stdout and hide its stderr.
   int ret;
   std::string stdOut;
