diff --git a/Source/cmcmd.cxx b/Source/cmcmd.cxx
index d05e3c8..22f649b 100644
--- a/Source/cmcmd.cxx
+++ b/Source/cmcmd.cxx
@@ -212,6 +212,12 @@ static int HandleTidy(const std::string& runCmd, const std::string& sourceFile,
   tidy_cmd.emplace_back("--");
   cmAppend(tidy_cmd, orig_cmd);
 
+  std::string target_sys;
+  if (cmSystemTools::GetEnv("TARGET_SYS", target_sys)) {
+    tidy_cmd.emplace_back("-target");
+    tidy_cmd.emplace_back(target_sys);
+  }
+
   // Run the tidy command line.  Capture its stdout and hide its stderr.
   std::string stdOut;
   std::string stdErr;
