diff --git a/Source/cmcmd.cxx b/Source/cmcmd.cxx
index 500f4874..2435d4db 100644
--- a/Source/cmcmd.cxx
+++ b/Source/cmcmd.cxx
@@ -447,7 +447,7 @@ int HandleCppLint(const std::string& runCmd, const std::string& sourceFile,
 
   // always return 0 so the build can continue as cpplint returns non-zero
   // for any warning
-  return 0;
+  return ret;
 }
 
 int HandleCppCheck(const std::string& runCmd, const std::string& sourceFile,
