diff --git a/src/FileTypeFactory.cpp b/src/FileTypeFactory.cpp
index 2088562..759f5c4 100644
--- a/src/FileTypeFactory.cpp
+++ b/src/FileTypeFactory.cpp
@@ -16,7 +16,7 @@ IFileTypePtr FileTypeFactory::CreateFileType(
     unsigned minChars) {
     auto ext = StringUtil::ToLower(StringUtil::GetFileExtension(filename));
     IFileTypePtr fileType;
-    if (ext == "c" || ext == "cpp" || ext == "cxx" || ext == "h" || ext == "cc")
+    if (ext == "c" || ext == "cpp" || ext == "cxx" || ext == "h" || ext == "cc" || ext == "hpp")
         fileType.reset(new FileType_C(ignorePrepStuff, minChars));
     else if (ext == "cs")
         fileType.reset(new FileType_CS(ignorePrepStuff, minChars));
