diff --git a/oelint_parser/cls_item.py b/oelint_parser/cls_item.py
index e3ae875..ee3280c 100644
--- a/oelint_parser/cls_item.py
+++ b/oelint_parser/cls_item.py
@@ -5,7 +5,7 @@ import os
 from oelint_parser.constants import CONSTANTS
 
 
-class Item():
+class Item(object):
     """Base class for all Stash items
     """
     ATTR_LINE = "Line"
@@ -270,7 +270,7 @@ class Variable(Item):
             operator {str} -- Operation performed to the variable
             flag {str} -- Optional variable flag
         """
-        super().__init__(origin, line, infileline, rawtext, realraw)
+        super(Variable, self).__init__(origin, line, infileline, rawtext, realraw)
         if "inherit" != name and not flag:
             self.__VarName, self.__SubItem = self.extract_sub(
                 name)
@@ -465,7 +465,7 @@ class Comment(Item):
             rawtext {str} -- Raw input string (except inline code blocks)
             realraw {str} -- Unprocessed input
         """
-        super().__init__(origin, line, infileline, rawtext, realraw)
+        super(Comment, self).__init__(origin, line, infileline, rawtext, realraw)
 
     def get_items(self):
         """Get single lines of block
@@ -493,7 +493,7 @@ class Include(Item):
             incname {str} -- raw name of the include file
             statement {str} -- either include or require
         """
-        super().__init__(origin, line, infileline, rawtext, realraw)
+        super(Include, self).__init__(origin, line, infileline, rawtext, realraw)
         self.__IncName = incname
         self.__Statement = statement
 
@@ -541,7 +541,7 @@ class Export(Item):
             name {str} -- variable name of the export
             value {str} -- (optional) value of the export
         """
-        super().__init__(origin, line, infileline, rawtext, realraw)
+        super(Export, self).__init__(origin, line, infileline, rawtext, realraw)
         self.__Name = name
         self.__Value = value
 
@@ -593,7 +593,7 @@ class Function(Item):
             python {bool} -- python function according to parser (default: {False})
             fakeroot {bool} -- uses fakeroot (default: {False})
         """
-        super().__init__(origin, line, infileline, rawtext, realraw)
+        super(Function, self).__init__(origin, line, infileline, rawtext, realraw)
         self.__IsPython = python is not None
         self.__IsFakeroot = fakeroot is not None
         name = name or ""
@@ -738,7 +738,7 @@ class PythonBlock(Item):
             realraw {str} -- Unprocessed input
             name {str} -- Function name
         """
-        super().__init__(origin, line, infileline, rawtext, realraw)
+        super(PythonBlock, self).__init__(origin, line, infileline, rawtext, realraw)
         self.__FuncName = name
 
     @property
@@ -778,7 +778,7 @@ class TaskAssignment(Item):
             ident {str} -- task flag
             value {str} -- value of modification
         """
-        super().__init__(origin, line, infileline, rawtext, realraw)
+        super(TaskAssignment, self).__init__(origin, line, infileline, rawtext, realraw)
         self.__FuncName = name
         self.__VarName = ident
         self.__VarValue = value
@@ -833,7 +833,7 @@ class FunctionExports(Item):
             realraw {str} -- Unprocessed input
             name {str} -- name of function to be exported
         """
-        super().__init__(origin, line, infileline, rawtext, realraw)
+        super(FunctionExports, self).__init__(origin, line, infileline, rawtext, realraw)
         self.__FuncNames = name
 
     @property
@@ -883,7 +883,7 @@ class TaskAdd(Item):
             before {str} -- before statement (default: {""})
             after {str} -- after statement (default: {""})
         """
-        super().__init__(origin, line, infileline, rawtext, realraw)
+        super(TaskAdd, self).__init__(origin, line, infileline, rawtext, realraw)
         self.__FuncName = name
         self.__Before = [x for x in (before or "").split(" ") if x]
         self.__After = [x for x in (after or "").split(" ") if x]
@@ -939,7 +939,7 @@ class MissingFile(Item):
             filename {str} -- filename of the file that can't be found
             statement {str} -- either include or require
         """
-        super().__init__(origin, line, infileline, "", "")
+        super(MissingFile, self).__init__(origin, line, infileline, "", "")
         self.__Filename = filename
         self.__Statement = statement
 
diff --git a/oelint_parser/cls_stash.py b/oelint_parser/cls_stash.py
index 098210e..735fcb4 100644
--- a/oelint_parser/cls_stash.py
+++ b/oelint_parser/cls_stash.py
@@ -7,7 +7,7 @@ from oelint_parser.helper_files import expand_term, guess_recipe_name, guess_rec
 from oelint_parser.constants import CONSTANTS
 
 
-class Stash():
+class Stash(object):
 
     def __init__(self, quiet=False):
         """constructor
@@ -182,7 +182,7 @@ class Stash():
             "PV": guess_recipe_version(filename),
             "BPN": guess_base_recipe_name(filename)
         }
-        _exp = {**_exp, **CONSTANTS.SetsBase}
+        _exp = dict(list(_exp.items()) + list(CONSTANTS.SetsBase.items()))
         for item in sorted(_res, key=lambda x: x.Line):
             varop = item.VarOp
             name = item.VarNameComplete
diff --git a/oelint_parser/constants.py b/oelint_parser/constants.py
index f84974b..1670553 100644
--- a/oelint_parser/constants.py
+++ b/oelint_parser/constants.py
@@ -26,7 +26,7 @@ class Constants():
         try:
             with open(path) as _in:
                 return json.load(_in)
-        except (FileNotFoundError, OSError, json.JSONDecodeError) as e:
+        except (IOError, OSError, json.JSONDecodeError) as e:
             sys.stderr.write('Cannot load constant database\n')
             return {}
 
diff --git a/oelint_parser/helper_files.py b/oelint_parser/helper_files.py
index ef00da2..7fa8036 100644
--- a/oelint_parser/helper_files.py
+++ b/oelint_parser/helper_files.py
@@ -1,7 +1,10 @@
 import glob
 import os
 import re
-from urllib.parse import urlparse
+try:
+    from urllib.parse import urlsplit
+except ImportError:
+    from urlparse import urlsplit
 
 from oelint_parser.cls_item import Variable
 from oelint_parser.constants import CONSTANTS
@@ -93,7 +96,7 @@ def get_scr_components(string):
         dict -- scheme: protocol used, src: source URI, options: parsed options
     """
     _raw = _replace_with_known_mirrors(string)
-    _url = urlparse(_replace_with_known_mirrors(string))
+    _url = urlsplit(_replace_with_known_mirrors(string))
     _scheme = _url.scheme
     _tmp = _url.netloc
     if _url.path:
diff --git a/oelint_parser/parser.py b/oelint_parser/parser.py
index eb3cf2c..e415da0 100644
--- a/oelint_parser/parser.py
+++ b/oelint_parser/parser.py
@@ -60,13 +60,13 @@ def prepare_lines_subparser(_iter, lineOffset, num, line, raw_line=None):
     res = []
     raw_line = raw_line or line
     if re.search(r"\\\s*\n", raw_line):
-        _, line = _iter.__next__()
+        _, line = next(_iter)
         while re.search(r"\\\s*\n", line):
             raw_line += line
-            _, line = _iter.__next__()
+            _, line = next(_iter)
         raw_line += line
     elif re.match(__func_start_regexp__, raw_line):
-        _, line = _iter.__next__()
+        _, line = next(_iter)
         stopiter = False
         scope_level = 0
         while not stopiter:
@@ -76,7 +76,7 @@ def prepare_lines_subparser(_iter, lineOffset, num, line, raw_line=None):
             if "}" in line:
                 scope_level -= 1
             try:
-                _, line = _iter.__next__()
+                _, line = next(_iter)
             except StopIteration:
                 stopiter = True
             if line.strip() == "}" and not scope_level:
@@ -87,7 +87,7 @@ def prepare_lines_subparser(_iter, lineOffset, num, line, raw_line=None):
         stopiter = False
         while not stopiter:
             try:
-                _, line = _iter.__next__()
+                _, line = next(_iter)
             except StopIteration:
                 stopiter = True
             if re.match("^[A-Za-z0-9#]+", line) or stopiter:
@@ -131,7 +131,7 @@ def prepare_lines(_file, lineOffset=0):
             for num, line in _iter:
                 prep_lines += prepare_lines_subparser(
                     _iter, lineOffset, num, line)
-    except FileNotFoundError:
+    except (IOError, OSError):
         pass
     return prep_lines
 
diff --git a/setup.py b/setup.py
index 3c51617..9fba916 100644
--- a/setup.py
+++ b/setup.py
@@ -7,7 +7,7 @@ try:
     _long_description = subprocess.check_output(
         ["pandoc", "--from", "markdown", "--to", "markdown", "README.md"]).decode("utf-8")
     _long_description_content_type = "text/markdown"
-except (subprocess.CalledProcessError, FileNotFoundError):
+except (subprocess.CalledProcessError, IOError, OSError):
     pass
 
 requirements = []
