diff --git a/oelint_parser/parser.py b/oelint_parser/parser.py
index a21d1e8..11252a7 100644
--- a/oelint_parser/parser.py
+++ b/oelint_parser/parser.py
@@ -139,7 +139,7 @@ def prepare_lines(_file, lineOffset=0):
     return prep_lines
 
 
-def get_items(stash, _file, lineOffset=0):
+def get_items(stash, _file, lineOffset=0, addInheritedFile=False, addIncludedFile=True):
     """parses file
 
     Args:
@@ -216,21 +216,22 @@ def get_items(stash, _file, lineOffset=0):
                     good = True
                     break
                 elif k == "inherit":
-                    inhname = expand_term(stash, _file, m.group("inhname"))
-                    if not inhname.endswith(".bbclass"):
-                        inhname += ".bbclass"
-                    _path = None
-                    for location in ["classes", "classes-recipe", "classes-global"]:
-                        _path = find_local_or_in_layer(
-                            os.path.join(location, inhname),
-                            os.path.dirname(_file))
+                    if addInheritedFile:
+                        inhname = expand_term(stash, _file, m.group("inhname"))
+                        if not inhname.endswith(".bbclass"):
+                            inhname += ".bbclass"
+                        _path = None
+                        for location in ["classes", "classes-recipe", "classes-global"]:
+                            _path = find_local_or_in_layer(
+                                os.path.join(location, inhname),
+                                os.path.dirname(_file))
+                            if _path:
+                                break
                         if _path:
-                            break
-                    if _path:
-                        tmp = stash.AddFile(
-                            _path, lineOffset=line["line"], forcedLink=_file)
-                        if any(tmp):
-                            includeOffset += max([x.InFileLine for x in tmp])
+                            tmp = stash.AddFile(
+                                _path, lineOffset=line["line"], forcedLink=_file)
+                            if any(tmp):
+                                includeOffset += max([x.InFileLine for x in tmp])
                     res.append(Variable(
                         _file, line["line"] + includeOffset, line["line"] -
                         lineOffset, line["raw"], "inherit", m.group(
@@ -272,13 +273,14 @@ def get_items(stash, _file, lineOffset=0):
                         _file, line["line"] + includeOffset, line["line"] - lineOffset, line["raw"], m.group("func"), line["realraw"], _b, _a))
                     break
                 elif k == "include":
-                    _path = find_local_or_in_layer(
-                        expand_term(stash, _file, m.group("incname")), os.path.dirname(_file))
-                    if _path:
-                        tmp = stash.AddFile(
-                            _path, lineOffset=line["line"], forcedLink=_file)
-                        if any(tmp):
-                            includeOffset += max([x.InFileLine for x in tmp])
+                    if addIncludedFile:
+                        _path = find_local_or_in_layer(
+                            expand_term(stash, _file, m.group("incname")), os.path.dirname(_file))
+                        if _path:
+                            tmp = stash.AddFile(
+                                _path, lineOffset=line["line"], forcedLink=_file)
+                            if any(tmp):
+                                includeOffset += max([x.InFileLine for x in tmp])
                     res.append(Include(
                         _file, line["line"], line["line"] - lineOffset, line["raw"], m.group("incname"), m.group("statement"), line["realraw"]))
                     good = True
