diff --git a/setup.py b/setup.py
index ea022e2..91e56ae 100755
--- a/setup.py
+++ b/setup.py
@@ -36,10 +36,6 @@ class Lint(Cmd):
         self.execute('cpplint.py')
 
 
-with open('test-requirements') as f:
-    test_required = f.read().splitlines()
-
-
 setup(name='cpplint',
       version=cpplint.__VERSION__,
       py_modules=['cpplint'],
@@ -71,14 +67,9 @@ setup(name='cpplint',
       description='Automated checker to ensure C++ files follow Google\'s style guide',
       long_description=open('README.rst').read(),
       license='BSD-3-Clause',
-      setup_requires=[
-          "pytest-runner"
-      ],
-      tests_require=test_required,
       # extras_require allow pip install .[dev]
       extras_require={
-          'test': test_required,
-          'dev': open('dev-requirements').read().splitlines() + test_required
+          'dev': open('dev-requirements').read().splitlines()
       },
       cmdclass={
           'lint': Lint
