diff --git a/cpplint.py b/cpplint.py
index 6b78b30..53f6ee8 100755
--- a/cpplint.py
+++ b/cpplint.py
@@ -6531,7 +6531,7 @@ def ProcessConfigOverrides(filename):
       continue
 
     try:
-      with open(cfg_file, encoding='utf-8') as file_handle:
+      with codecs.open(cfg_file, 'r', 'utf8', 'replace') as file_handle:
         for line in file_handle:
           line, _, _ = line.partition('#')  # Remove comments.
           if not line.strip():
