Upstream-Status: Pending

diff --git a/oelint_adv/__main__.py b/oelint_adv/__main__.py
index d4c9b0a..b59460d 100644
--- a/oelint_adv/__main__.py
+++ b/oelint_adv/__main__.py
@@ -4,6 +4,8 @@ import multiprocessing as mp
 import os
 import re
 import sys
+import logging
+
 from configparser import ConfigParser
 from configparser import NoOptionError
 from configparser import NoSectionError
@@ -30,6 +32,8 @@ from oelint_adv.version import __version__
 
 sys.path.append(os.path.abspath(os.path.join(__file__, '..')))
 
+logger = logging.getLogger('oelint-adv')
+
 
 class TypeSafeAppendAction(argparse.Action):
 
@@ -376,6 +380,9 @@ def make_json_report(issues):
 
 
 def main():  # pragma: no cover
+    logging.basicConfig(stream=sys.stdout, level=logging.INFO)
+    logger.info("Checking the specified recipes or files for the styling issues...")
+
     args = arguments_post(parse_arguments())
 
     if args.print_rulefile:
@@ -393,6 +400,8 @@ def main():  # pragma: no cover
     if issues:
         sys.stdout.write('\n')
 
+    logger.info("Done.")
+
     exit_code = len(issues) if not args.exit_zero else 0
     sys.exit(exit_code)
 
