diff --git a/oelint_adv/__main__.py b/oelint_adv/__main__.py
index 56d7423..731f69a 100644
--- a/oelint_adv/__main__.py
+++ b/oelint_adv/__main__.py
@@ -3,6 +3,7 @@ import json
 import os
 import re
 import sys
+import logging
 
 try:
     from configparser import ConfigParser
@@ -29,6 +30,8 @@ from oelint_adv.rule_file import set_suppressions
 
 sys.path.append(os.path.abspath(os.path.join(__file__, '..')))
 
+logger = logging.getLogger('oelint-adv')
+
 
 class TypeSafeAppendAction(argparse.Action):
 
@@ -317,6 +320,9 @@ def make_json_report(issues):
 
 
 def main():  # pragma: no cover
+    logging.basicConfig(stream=sys.stdout, level=logging.INFO)
+    logger.info("Checking the specified recipes or files for the styling issues...")
+
     args = arguments_post(parse_arguments())
 
     if args.print_rulefile:
@@ -334,6 +340,8 @@ def main():  # pragma: no cover
     if issues:
         sys.stdout.write('\n')
 
+    logger.info("Done.")
+
     exit_code = len(issues) if not args.exit_zero else 0
     sys.exit(exit_code)
 
