diff --git a/oelint_adv/rule_base/rule_var_homepageping.py b/oelint_adv/rule_base/rule_var_homepageping.py
index 3e40134..e8535fa 100644
--- a/oelint_adv/rule_base/rule_var_homepageping.py
+++ b/oelint_adv/rule_base/rule_var_homepageping.py
@@ -12,6 +12,8 @@ except ImportError:
 from oelint_adv.cls_rule import Rule
 from oelint_parser.cls_item import Variable
 
+from ssl import _create_unverified_context
+
 
 class VarHomepagePing(Rule):
     def __init__(self):
@@ -27,7 +29,7 @@ class VarHomepagePing(Rule):
             try:
                 req = Request(i.VarValueStripped)  # noqa: S310 - we can take the risk of calling unexpected schemes here
                 try:
-                    urlopen(req, timeout=4)  # noqa: S310 - we can take the risk of calling unexpected schemes here
+                    urlopen(req, timeout=4, context=_create_unverified_context())  # noqa: S310 - we can take the risk of calling unexpected schemes here
                 except HTTPError as e:
                     if e.code == 404:  # pragma: no cover
                         res += self.finding(i.Origin,
