diff --git a/oelint_adv/rule_base/rule_var_homepageping.py b/oelint_adv/rule_base/rule_var_homepageping.py
index ddf8c2f..755f378 100644
--- a/oelint_adv/rule_base/rule_var_homepageping.py
+++ b/oelint_adv/rule_base/rule_var_homepageping.py
@@ -1,6 +1,8 @@
 from oelint_parser.cls_item import Variable
 from oelint_adv.cls_rule import Rule
 
+from ssl import _create_unverified_context
+
 try:
     from urllib.request import urlopen, Request
     from urllib.error import HTTPError, URLError
@@ -22,8 +24,9 @@ class VarHomepagePing(Rule):
         for i in items:
             try:
                 req = Request(i.VarValueStripped)
+                context = _create_unverified_context()
                 try:
-                    urlopen(req, timeout=4)
+                    urlopen(req, timeout=4, context=context)
                 except HTTPError as e:
                     if e.code == 404: # pragma: no cover
                         res += self.finding(i.Origin, i.InFileLine) # pragma: no cover

