diff --git a/metrixpp/ext/std/tools/collect.py b/metrixpp/ext/std/tools/collect.py
index 3c4bd9a..c5f48e2 100644
--- a/metrixpp/ext/std/tools/collect.py
+++ b/metrixpp/ext/std/tools/collect.py
@@ -152,7 +152,8 @@ class DirectoryReader():
                     else:
                         logging.info("Processing: " + norm_path)
                         ts = time.time()
-                        f = open(full_path, 'rU');
+                        import io
+                        f = io.open(full_path, 'rU', encoding='utf-8', errors='ignore');
                         text = f.read();
                         # getting along with the different string handling of python 2 and 3
                         # trying to get along with different encodings to get the tests running
